### DISCUSSION TITLE
Pagination support

### DISCUSSION THREAD

**Discussion by danielloader** [grafana/grafana-json-datasource#254] [Ideas]
I know this is a difficult ask as every API has a different implementation of pagination, and I'm dealing with a particularly non standard implementation [here ](https://docs.pingdom.com/api/?_ga=2.230003480.509660209.1590495493-1793431897.1589990976#tag/Results/paths/~1results~1{checkid}/get)

The issue being I need to keep incrementing the `offset` value with the `to` and `from` parameters provided by Grafana, until the offset yields zero results (because its outside the time range). 

Understand it's a loose brief here so ideas are welcome (or marking it as not feasible). 

Thanks!

**Comment by marcusolsson** [comment]
Yeah, I've been thinking about this a few times and I'm not sure how to implement this other than to add support for the most common strategies. Also, currently the plugin maps a query to a single HTTP request. This would allow a single query to perform multiple queries in the background. That poses some challenges when it comes to caching, but also partial results. What if you hit a rate limit after the first 10 pages, with 10 more to go. Not insurmountable, just some thoughts that have come to mind.

**Comment by danielloader** [comment]
Definitely, I was hesitant to even open the ticket given every API does it differently, maybe some more input - even if it doesn't help me, could help others with this.

**Comment by userlerueda** [comment]
What about working with the "next" key in the body ? It could be a starting point. 
https://jsonapi.org/format/#fetching-pagination
